packages/flask/src/UICountryPicker/CountryPicker.tsx

```diff
@@ -0,0 +1,238 @@
+        }
+        return isPermitted && !isBanned;
+    }, []);
+
+    const filterCountries = React.useCallback(
+        list => {
+            const permittedCountries = list.filter((country: any) => checkIncludes(country.code));
+            setFilteredList(permittedCountries);
+        },
+        [banned, permitted],
+    );
+
+    React.useEffect(() => {
+        setFilteredList(countriesList);
+    }, [countriesList]);
+
+    React.useEffect(() => {
+        const result = search ? fuse.search(search) : countriesList;
+        setFilteredList(result as CountriesArray);
+    }, [search]);
+
+    React.useEffect(() => {
+        fetchJSON()
+            .then((r: CountriesArray) => {
+                if (permitted.length || banned.length) {
+                    filterCountries(r);
+                } else {
+                    setCountriesList(r);
+                }
+            })
+            .catch((e: Error) => {
+                console.error(e);
+            })
+            .finally(() => {
+                setLoading(false);
+            });
+    }, []);
+
+    const renderSearchHeader = () => {
+        return (
+            <View style={styles.headerContainer}>
+                <View style={styles.headerTitleContainer}>
+                    <View style={styles.sideHeaderView}>
+                        <UILinkButton onPress={onClose} title={uiLocalized.Cancel} />
+                    </View>
+                    <UILabel role={TypographyVariants.HeadlineHead} style={styles.headerTitle}>
+                        {uiLocalized.CountryPicker.Title}
+                    </UILabel>
+                    <View style={styles.sideHeaderView} />
+                </View>
+                <UISearchBar returnKeyType="done" value={search} onChangeText={setSearch} />
+            </View>
+        );
+    };
+
+    const renderCountryRow = ({ item }: { item: Country }) => {
+        const onPress = () => onSelectCountry(item);
+        return (
+            <TouchableOpacity onPress={onPress} style={styles.rowContainer}>
+                <View style={styles.rowContainerInner}>
+                    <UILabel>{item.name}</UILabel>
+                    <UILabel style={styles.emojiContainer}>{item.emoji}</UILabel>
+                </View>
+            </TouchableOpacity>
+        );
+    };
+
+    const renderLoading = () => {
+        return <ActivityIndicator />;
+    };
+
+    const renderEmptyList = () => {
+        return (
+            <>
+                <UILabel role={TypographyVariants.TitleMedium} color={ColorVariants.TextSecondary}>
+                    {uiLocalized.CountryPicker.CountryNotFindTitle}
+                </UILabel>
+                <UILabel role={TypographyVariants.Action} color={ColorVariants.TextSecondary}>
+                    {uiLocalized.CountryPicker.CountryNotFindDetails}
+                </UILabel>
+            </>
+        );
+    };
+
+    const ListEmptyComponent = React.useMemo(() => {
```

Comment:
What do you think to split it into 3 componenents?
- The wrapper that has `loading` in it
- The one for `ListEmptyComponent` to not re-create it on every render, and that gets loading state from context
- The one that renders a list

Reply:
Or it can be 2 components, with `ListEmptyComponent` getting `loading` from context